File: Java/Permutation Sequence.java
public class Solution {
    public String getPermutation(int n, int k) {
        ArrayList<Integer> nums = new ArrayList<Integer>();
        int[] lens = new int[n+1];
        lens[0] = 1;
        for (int i=1; i<=n; i++) {
            nums.add(i);
            lens[i] = lens[i-1]*i;
        }
        
        k -= 1;
        StringBuilder res = new StringBuilder();
        
        while (n > 1) {
            int len = lens[n-1];
            int num = nums.get(k/len);
            res.append(num);
            nums.remove(k/len);
            n -= 1;
            k = k%len;
        }
        
        res.append(nums.get(0));
        return res.toString();
    }
}

File: Python/Permutation Sequence.py
class Solution(object):
    """
    :type n: int
    :type k: int
    :rtype: str
    """
    def getPermutation(self, n, k):
        nums = []
        lens = []
        lens.append(1)
        for i in range(1, n+1, 1):
            nums.append(i)
            lens.append(lens[i-1]*i)
        
        k -= 1
        res = []
        
        while n > 1:
            len = lens[n-1]
            num = nums[k/len]
            res.append(str(num))
            del nums[k/len]
            n -= 1
            k %= len
        
        res.append(str(nums[0]))
        return "".join(res)
        